applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -60,17 +60,21 @@ void  ElasticIsotropic3D::CalculateMaterialResponsePK2(ConstitutiveLaw::Paramete
     KRATOS_TRY;
     // b.- Get Values to compute the constitutive law:
     Flags & r_constitutive_law_options = rValues.GetOptions();
-
     Vector& r_strain_vector = rValues.GetStrainVector();
 
     //NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
     if( r_constitutive_law_options.IsNot( ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN )) {
         CalculateCauchyGreenStrain( rValues, r_strain_vector);
     }
 
+    const Vector& r_initial_strain = this->GetInitialStrainVector(rValues);
+    r_strain_vector -= r_initial_strain;
```

Comment:
it would be more efficient to check here if initial stress/strain was specified

Reply:
mm Why? the initial strain or stress can be imposed separatedly